Missing atomic fallback support/no libatomic

```C
#include <stdint.h>

int main() {
    __uint8_t value = 0x1;
    __uint128_t target = 0;

    __atomic_store_16(&target, value, __ATOMIC_SEQ_CST);
}
```

```
root@9c84f874efb9 /mnt# x86_64-hermit-gcc -o main main.c -L/mnt -latomic
/opt/hermit/lib/gcc/x86_64-hermit/14.2.0/../../../../x86_64-hermit/bin/ld: cannot find -latomic: No such file or directory
collect2: error: ld returned 1 exit status
```

--- a/libatomic/configure.tgt
+++ b/libatomic/configure.tgt
@@ -151,7 +151,7 @@ case "${target}" in
   *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu \
   | *-*-netbsd* | *-*-freebsd* | *-*-openbsd* | *-*-dragonfly* \
   | *-*-solaris2* | *-*-sysv4* | *-*-irix6* | *-*-osf* | *-*-hpux11* \
-  | *-*-darwin* | *-*-aix* | *-*-cygwin*)
+  | *-*-darwin* | *-*-aix* | *-*-cygwin* | *-*-hermit*)
 	# POSIX system.  The OS is supported.
 	config_path="${config_path} posix"
 	;;

